python/paddle/v2/fluid/optimizer.py

```diff
@@ -36,10 +36,18 @@ class Optimizer(object):
     """
 
     def __init__(self, learning_rate, global_step=None, regularization=None):
-        assert learning_rate is not None
+        if not isinstance(learning_rate, float) and \
+                not isinstance(learning_rate, framework.Variable):
+            raise ValueError("learning rate should be float or Variable")
         self._global_step = global_step
         self.regularization = regularization
-        self._global_learning_rate = learning_rate
+        self._learning_rate = learning_rate
+        # each program should have a independent learning rate
+        # program -> Variable(learning_rate)
+        self._learning_rate_map = defaultdict(lambda: None)
```

Comment:
There is an API called `get` in `dict`.
<img width="876" alt="default" src="https://user-images.githubusercontent.com/728699/36650515-e269493c-1ade-11e8-97ed-534d0d165208.png">

We can use that API rather than defaultdict.

Reply:
done